compiler/rustc_infer/src/infer/error_reporting/mod.rs

```diff
@@ -1428,6 +1428,7 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
         secondary_span: Option<(Span, String)>,
         mut values: Option<ValuePairs<'tcx>>,
         terr: &TypeError<'tcx>,
+        swap_secondary_and_primary: bool,
```

Comment:
hmm, should this have a comment with some guidance about when the `swap_secondary_and_primary` argument should be `true`? Or I guess we can rely on people just trying it out and seeing how it looks?

Reply:
Sure, I can add a docstring mentioning the function, but I would expect that "see how it affects the output" would be _my_ natural inclination.

Reply:
Added a docstring to the method with some info.